src/test/java/bio/terra/stairway/FlightMapTest.java

```diff
@@ -0,0 +1,57 @@
+package bio.terra.stairway;
+
+import bio.terra.stairway.fixtures.FlightsTestPojo;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Tag("unit")
+public class FlightMapTest {
+
+  private static final Logger logger = LoggerFactory.getLogger("FlightMapTest");
+
+  @Test
+  public void toAndFromJson() {
+    FlightMap outMap = new FlightMap();
+
+    FlightsTestPojo pojoIn = new FlightsTestPojo().anint(99).astring("mystring");
+    outMap.put("mypojo", pojoIn);
+
+    Integer intIn = 3;
+    outMap.put("myint", intIn);
+
+    String json = outMap.toJson();
+    logger.info("JSON: '{}'", json);
+
+    FlightMap inMap = new FlightMap();
+    inMap.fromJson(json);
+
+    FlightsTestPojo pojoOut = inMap.get("mypojo", FlightsTestPojo.class);
+    Assertions.assertEquals(pojoIn, pojoOut);
+
+    Integer intOut = inMap.get("myint", Integer.class);
+    Assertions.assertEquals(intIn, intOut);
+  }
+
+  @Test
+  public void fromJsonMap() {
+    FlightMap map = new FlightMap();
+
+    String jsonMap =
+        "[\"java.util.HashMap\",{\"mykey\":[\"bio.terra.stairway.fixtures.FlightsTestPojo\",{\"astring\":\"mystring\",\"anint\":99}]}]";
+
```

Comment:
Maybe we could harvest a bunch of samples from the TDR dev stairway database and run them through this logic to see if we get any surprises.

Reply:
I like that idea, though test will need to be written in TDR (and might be less useful now that we've reverted most of FlightMap).

Reply:
Yes, true. It wouldn't work. I think this is fine.